lib/marc_match_key/key.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+### A collection of functions and classes to generate a match key based on the
+###   GoldRush algorithm
+module MarcMatchKey
+  ### Generates the Gov Doc portion of a GoldRush key
+  class Key
+    attr_reader :record
+
+    def initialize(record)
+      @record = record
+    end
+
+    def key
+      [key_part_one, key_part_two, key_part_three].join
+    end
+
+    private
+
+    def key_part_one
+      part = TitleKey.new(record).key
+      part << PubDateKey.new(record).key
+      part << PaginationKey.new(record).key
+      part << EditionKey.new(record).key
```

Comment:
Maybe we could also use the array/join approach for `#key_part_one`, `#key_part_two`, and `#key_part_three`?  Something like:

```
    def key_part_one
      [TitleKey.new(record).key, PubDateKey.new(record).key, PaginationKey.new(record).key, EditionKey.new(record).key].join
    end
```

Or even:

```
    def key_part_one
      [TitleKey, PubDateKey, PaginationKey, EditionKey]
        .map {|key_type| key_type.new(record).key}
        .join
    end
```

Reply:
I was looking for something like the second proposal!